jax/_src/tree_util.py

```diff
@@ -27,6 +28,12 @@
 T = TypeVar("T")
 U = TypeVar("U")
 
+if TYPE_CHECKING:
```

Comment:
Any reason not to do this unconditionally?

Reply:
Yep, `PyTreeDef` is not exported by pybind11. I think this is because because it has no exported `__init__`.

```
>>> from jaxlib.xla_extension import pytree
>>> dir(pytree)
['__doc__', '__loader__', '__name__', '__package__', '__spec__', 'all_leaves', 'flatten', 'register_node', 'tuple']
```

Reply:
Seems like something we should fix, right?

Reply:
Sure, I'll look into this in a separate PR.

Reply:
Addressing this in #11455